tck/src/main/java/org/eclipse/microprofile/config/tck/ArrayConverterTest.java

```diff
@@ -1654,120 +1658,161 @@ public void testInstantSetInjection() {
     }
 
     @Test
-    public void testOptionalUrlListLookupProgrammatically() throws MalformedURLException {
+    public void testOptionalUrlListLookupProgrammatically() throws MalformedURLException, URISyntaxException {
         Optional<List<URL>> optionalValues = config.getOptionalValues("tck.config.test.javaconfig.converter.urlvalues", URL.class);
         Assert.assertTrue(optionalValues.isPresent());
         List<URL> values = optionalValues.get();
         Assert.assertNotNull(values);
         Assert.assertEquals(values.size(), 3);
-        Assert.assertEquals(values, Arrays.asList(
-            new URL("http://microprofile.io"),
-            new URL("http://openliberty.io"),
-            new URL("http://microprofile.io")));
+        assertURLListEquals(values);
         Optional<List<URL>> optionalSingle = config.getOptionalValues("tck.config.test.javaconfig.converter.urlvalue", URL.class);
         Assert.assertTrue(optionalSingle.isPresent());
         List<URL> single = optionalSingle.get();
         Assert.assertNotNull(single);
         Assert.assertEquals(single.size(), 1);
-        Assert.assertEquals(single, Arrays.asList(new URL("http://microprofile.io")));
+        Assert.assertEquals(single.get(0).toURI(), Arrays.asList(new URL("http://microprofile.io")).get(0).toURI());
     }
 
 
     @Test
-    public void testUrlArrayInjection() throws MalformedURLException {
-        Assert.assertEquals(converterBean.getMyUrlArray().length, 3);
-        Assert.assertEquals(converterBean.getMyUrlArray(), new URL[]{
-            new URL("http://microprofile.io"),
-            new URL("http://openliberty.io"),
-            new URL("http://microprofile.io")});
+    public void testUrlArrayInjection() throws MalformedURLException, URISyntaxException {
+        URL[] values = converterBean.getMyUrlArray();
+        Assert.assertEquals(values.length, 3);
+        assertURLArrayEquals(values);
         Assert.assertEquals(converterBean.getMySingleUrlArray().length, 1);
-        Assert.assertEquals(converterBean.getMySingleUrlArray(), new URL[]{ new URL("http://microprofile.io") });
+        Assert.assertEquals(converterBean.getMySingleUrlArray()[0].toURI(), new URL[]{ new URL("http://microprofile.io") }[0].toURI());
     }
 
     @Test
-    public void testURLListInjection() throws MalformedURLException {
-        Assert.assertEquals(converterBean.getMyUrlList().size(), 3);
-        Assert.assertEquals(converterBean.getMyUrlList(), Arrays.asList(
-            new URL("http://microprofile.io"),
+    public void testURLListInjection() throws MalformedURLException, URISyntaxException {
+        List<URL> values = converterBean.getMyUrlList();
+        Assert.assertEquals(values.size(), 3);
+        assertURLListEquals(values);
+        Assert.assertEquals(converterBean.getMySingleUrlList().size(), 1);
+        Assert.assertEquals(converterBean.getMySingleUrlList().get(0).toURI(), Arrays.asList(new URL("http://microprofile.io")).get(0).toURI());
```

Comment:
unnecessary list wrapping again, maybe it is intentional? I find it more confusing.

Reply:
refactored.